src/WebJobs.Extensions.NotificationHubs/Config/Converter.cs

```diff
@@ -9,19 +10,63 @@ namespace Microsoft.Azure.WebJobs.Extensions.NotificationHubs
 {
     internal static class Converter
     {
+        public static string NotificationPlatform { get; set; }
+
         public static IConverterManager AddNotificationHubConverters(this IConverterManager converterManager)
         {
             converterManager.AddConverter<TemplateNotification, Notification>(templateNotification => templateNotification);
-            converterManager.AddConverter<string, Notification>(messageProperties => BuildTemplateNotificationFromJsonString(messageProperties));
+            converterManager.AddConverter<string, Notification>(notificationAsString => BuildNotificationFromString(notificationAsString));
             converterManager.AddConverter<IDictionary<string, string>, Notification>(messageProperties => BuildTemplateNotificationFromDictionary(messageProperties));
             return converterManager;
         }
 
-        internal static TemplateNotification BuildTemplateNotificationFromJsonString(string messageProperties)
+        internal static Notification BuildNotificationFromString(string notificationAsString)
         {
-            JObject jobj = JObject.Parse(messageProperties);
-            Dictionary<string, string> templateProperties = jobj.ToObject<Dictionary<string, string>>();
-            return new TemplateNotification(templateProperties);
+            Notification notification = null;
+            if (string.IsNullOrEmpty(NotificationPlatform))
+            {
+                JObject jobj = JObject.Parse(notificationAsString);
+                Dictionary<string, string> templateProperties = jobj.ToObject<Dictionary<string, string>>();
+                notification = new TemplateNotification(templateProperties);
+            }
+            else
+            {
+                NotificationPlatform platform;
+                try
+                {
+                    platform = (NotificationPlatform)Enum.Parse(typeof(NotificationPlatform), NotificationPlatform);
```

Comment:
This would be set via bindings.json -- so this shouldn't affect user scripts. The only issue is whether the bindings.json is deserialized properly. If not, we can make it work. It'd be nicer to just use the enum here and remove all of your parsing code.